lib/signaling/v2/peerconnection.js

```diff
@@ -486,33 +487,90 @@ class PeerConnectionV2 extends StateMachine {
-    });
+  _maybeUpdateEncodings(track, encodings, trackReplaced = false) {
+    if (track.kind !== 'video') {
+      return false;
+    }
+    const browser = util.guessBrowser();
+
+    // Note(mpatwardhan): always configure encodings  always for safari.
+    // for chrome only when adaptive simulcast enabled.
+    if (browser === 'safari' || (browser === 'chrome' && this._isAdaptiveSimulcastEnabled())) {
+      return this._updateEncodings(track, encodings, trackReplaced);
+    }
+    return false;
+  }
+
+  /**
+   * Configures with default encodings depending on track type and resolution.
+   * Default configuration sets some encodings to disabled, and for others set scaleResolutionDownBy
+   * values. When trackReplaced is set to true, it will clear 'active' for any encodings that
+   * needs to be enabled.
+   * @param {MediaStreamTrack} track
+   * @param {Array<RTCRtpEncodingParameters>} encodings
+   * @param {boolean} trackReplaced
+   */
+  _updateEncodings(track, encodings, trackReplaced) {
+    if (this._isChromeScreenShareTrack(track)) {
+      // for screen share track only enable 2 layers.
+      const screenShareActiveLayerConfig = [
+        { scaleResolutionDownBy: 1 },
+        { scaleResolutionDownBy: 1 }
+      ];
+      encodings.forEach((encoding, i) => {
+        const activeLayerConfig = screenShareActiveLayerConfig[i];
+        if (activeLayerConfig) {
+          encoding.scaleResolutionDownBy = activeLayerConfig.scaleResolutionDownBy;
+          if (trackReplaced) {
+            delete encoding.active;
+          }
+        } else {
+          encoding.active = false;
+          delete encoding.scaleResolutionDownBy;
+        }
+      });
+    } else {
+      const { width, height }  = track.getSettings();
+      // NOTE(mpatwardhan): for non-screen share tracks
+      // enable layers depending on track resolutions
+      const pixelsToMaxActiveLayers = [
+        { pixels: 960 * 540, maxActiveLayers: 3 },
+        { pixels: 480 * 270, maxActiveLayers: 2 },
+        { pixels: 0, maxActiveLayers: 1 }
+      ];
+
+      const trackPixels =  width * height;
+      const activeLayersInfo = pixelsToMaxActiveLayers.find(layer => trackPixels >= layer.pixels);
+      const activeLayers = Math.min(encodings.length, activeLayersInfo.maxActiveLayers);
+      encodings.forEach((encoding, i) => {
+        const enabled  = i < activeLayers;
+        if (enabled) {
+          encoding.scaleResolutionDownBy = 1 << (activeLayers - i - 1);
+          if (trackReplaced) {
+            delete encoding.active;
+          }
+        } else {
+          encoding.active = false;
+          delete encoding.scaleResolutionDownBy;
+        }
+      });
+    }
+    this._log.debug('_updateEncodings:', encodings.map(({ active, scaleResolutionDownBy }, i) => `[${i}: ${active}, ${scaleResolutionDownBy || 0}]`).join(', '));
+    return true;
```

Comment:
`_updateEncodings` will always return true correct? Maybe remove the return type here and just return true where we're calling it inside `_maybeUpdateEncodings`?